mlfinlab/backtest_statistics/backtests.py

```diff
@@ -0,0 +1,544 @@
+        t_sample = self._sample_random_multest(parameters[0], num_trails, parameters[2], parameters[3],
+                                               self.simulations)
+
+        # Annual Sharpe ratio, adjusted to monthly
+        sr_monthly = sr_annual / 12 ** (1 / 2)
+        # Calculating t-ratio based on the Sharpe ratio and the number of observations
+        t_ratio = sr_monthly * monthly_obs ** (1 / 2)
+        # Calculating adjusted p-value from the given t-ratio
+        p_val = 2 * (1 - ss.t.cdf(t_ratio, monthly_obs - 1))
+
+        # Creating arrays for p-values from simulations of Holm and BHY methods.
+        p_holm = np.ones(self.simulations)
+        p_bhy = np.ones(self.simulations)
+
+        # Iterating through the simulations
+        for simulation_number in range(1, self.simulations + 1):
+
+            # Get one sample of previously generated simulation of t-values
+            t_values_simulation = t_sample[simulation_number - 1, 1:(num_mult_test + 1)]
+            # Calculating adjusted p-values from the simulated t-ratios
+            p_values_simulation = 2 * (1 - ss.norm.cdf(t_values_simulation, 0, 1))
+
+            # To the N (num_mult_test) other strategies tried (from the simulation),
+            # we add the adjusted p_value of the real strategy.
+            all_p_values = np.append(p_values_simulation, p_val)
+            # Ordering p-values
+            all_p_values = np.sort(all_p_values)
+
+            # Holm method
+            p_holm[simulation_number - 1] = self._holm_method_sharpe(all_p_values, num_mult_test, p_val)
+
+            # BHY method
+            p_bhy[simulation_number - 1] = self._bhy_method_sharpe(all_p_values, num_mult_test, p_val)
+
+        # Calculating the resulting p-values of methods from simulations
+        # Array with adjusted p-values
+        # [Bonferroni, Holm, BHY, Average]
+        p_val_adj = np.array([np.minimum(num_mult_test * p_val, 1), np.median(p_holm), np.median(p_bhy)])
+        p_val_adj = np.append(p_val_adj, (p_val_adj[0] + p_val_adj[1] + p_val_adj[2]) / 3)
+
+        # Arrays with adjusted Sharpe ratios and haircuts
+        sr_adj = np.zeros(4)
+        haircut = np.zeros(4)
+        # Adjusted Sharpe ratios and haircut percentages
+        sr_adj[0], haircut[0] = self._sharpe_ratio_haircut(p_val_adj[0], monthly_obs, sr_annual)
+        sr_adj[1], haircut[1] = self._sharpe_ratio_haircut(p_val_adj[1], monthly_obs, sr_annual)
+        sr_adj[2], haircut[2] = self._sharpe_ratio_haircut(p_val_adj[2], monthly_obs, sr_annual)
+        sr_adj[3], haircut[3] = self._sharpe_ratio_haircut(p_val_adj[3], monthly_obs, sr_annual)
+
+        results = np.array([p_val_adj,
+                            sr_adj,
+                            haircut])
+
+        return results
+
+    def profit_hurdle(self, num_mult_test, num_obs, alpha_sig, vol_anu, rho):
+        # pylint: disable=too-many-locals
+        '''
```

Comment:
All these docstrings should please be with double quotes and not single quotes. 

"""

\"""

Reply:
Will change